src/log/log.c

```diff
@@ -2481,29 +2482,49 @@ __wt_log_scan(WT_SESSION_IMPL *session, WT_LSN *lsnp, uint32_t flags,
 			 */
 			if (log != NULL)
 				log->trunc_lsn = rd_lsn;
-			/* Make a check to see if it may be a partial write. */
-			if (!__log_check_partial_write(session, buf, reclen)) {
-				/*
-				 * It's not a partial write, and we have a bad
-				 * checksum. We treat it as a corruption that
-				 * must be salvaged.
-				 */
-				need_salvage = true;
-				WT_ERR(__log_salvageable_error(session,
-				    log_fh->name, ", bad checksum",
-				    rd_lsn.l.offset));
-			} else {
-				/*
-				 * It may be a partial write, or it's possible
-				 * that the header is corrupt.  Make a sanity
-				 * check of the log record header.
-				 */
-				WT_ERR(__log_record_verify(session, log_fh,
-				    rd_lsn.l.offset, logrec, &corrupt));
-				if (corrupt) {
+
+			/*
+			 * When we have a checksum mismatch, we determine
+			 * whether it may be the result of:
+			 *  1) some expected corruption that can occur during
+			 *     backups
+			 *  2) a partial write that can naturally occur when
+			 *     an application crashes
+			 *  3) some other corruption
+			 *
+			 * As the first and second happen commonly in normal
+			 * operations, we will silently truncate the log when
+			 * it occurs. The third will result in an error during
+			 * recovery, and requires salvage to fix.
+			 */
+			if (!F_ISSET(conn, WT_CONN_WAS_BACKUP)) {
+				if (!__log_check_partial_write(session, buf,
+				    reclen)) {
+					/*
+					 * It's not a partial write, and we
+					 * have a bad checksum. We treat it as
+					 * a corruption that must be salvaged.
+					 */
 					need_salvage = true;
 					WT_ERR(__log_salvageable_error(session,
-					    log_fh->name, "", rd_lsn.l.offset));
+					    log_fh->name, ", bad checksum",
+					    rd_lsn.l.offset));
+				} else {
+					/*
+					 * It may be a partial write, or it's
+					 * possible that the header is corrupt.
+					 * Make a sanity check of the log
+					 * record header.
+					 */
+					WT_ERR(__log_record_verify(session,
+					    log_fh, rd_lsn.l.offset, logrec,
+					    &corrupt));
+					if (corrupt) {
+						need_salvage = true;
+						WT_ERR(__log_salvageable_error(
+						    session, log_fh->name, "",
+						    rd_lsn.l.offset));
+					}
```

Comment:
My minor complaint is that it is a bit of a shame to have another layer of conditional in that code. It is pretty deep already. Maybe something like this:
```
if (!__log_checksum_mismatch(...)) {
    if (log != NULL)
        log->trunc_lsn = rd_lsn;
    if (LF_ISSET(WT_LOGSCAN_ONE))
        ret = WT_NOTFOUND;
    if (F_ISSET(conn, WT_CONN_WAS_BACKUP))
        break;
    if (!log_check_partial_write(...)) {
    } else { ...
    }
    break;
}
```
That helps one layer (if it works, I haven't thought too deeply about the line reordering correctness).

Reply:
@sueloverso, I like it. I'll make that change and test it out.  BTW, it has previously been failing at the first or second iteration of test/format on a specific config.  It ran cleanly for > 10 on the one you reviewed.